test/org/parosproxy/paros/core/scanner/VariantMultipartFormParametersUnitTest.java

```diff
@@ -0,0 +1,205 @@
+
+import java.util.List;
+
+import org.junit.Test;
+import org.parosproxy.paros.network.HttpHeader;
+import org.parosproxy.paros.network.HttpMalformedHeaderException;
+import org.parosproxy.paros.network.HttpMessage;
+
+/**
+ * Unit test for {@link VariantMultipartFormParameters}.
+ */
+public class VariantMultipartFormParametersUnitTest {
+	
+	private static final String CRLF = "\r\n";
+
+	@Test
+	public void shouldHaveParametersListEmptyByDefault() {
+		// Given
+		VariantMultipartFormParameters variant = new VariantMultipartFormParameters();
+		// When
+		List<NameValuePair> parameters = variant.getParamList();
+		// Then
+		assertThat(parameters, is(empty()));
+	}
+
+	@Test(expected = UnsupportedOperationException.class)
+	public void shouldNotAllowToModifyReturnedParametersList() {
+		// Given
+		VariantMultipartFormParameters variant = new VariantMultipartFormParameters();
+		// When
+		variant.getParamList().add(new NameValuePair(NameValuePair.TYPE_POST_DATA, "name", "fred", 1));
+		// Then = UnsupportedOperationException
+	}
+
+	@Test(expected = IllegalArgumentException.class)
+	public void shouldFailToExtractParametersFromUndefinedMessage() {
+		// Given
+		VariantMultipartFormParameters variant = new VariantMultipartFormParameters();
+		HttpMessage undefinedMessage = null;
+		// When
+		variant.setMessage(undefinedMessage);
+		// Then = IllegalArgumentException
+	}
+
+	@Test
+	public void shouldExtractParametersFromAllParts() {
+		// Given
+		VariantMultipartFormParameters variant = new VariantMultipartFormParameters();
+		// When
+		variant.setMessage(createMessage());
+		// Then
+		assertThat(variant.getParamList().size(), is(equalTo(4)));
+		assertThat(variant.getParamList().get(0).getName(), is(equalTo("person")));
+		assertThat(variant.getParamList().get(0).getValue(), is(equalTo("anonymous" + CRLF + "anonymous2" + CRLF)));
+		assertThat(variant.getParamList().get(1).getName(), is(equalTo("somefile")));
+		assertThat(variant.getParamList().get(1).getValue(), is(equalTo("contents of the file" + CRLF)));
+		assertThat(variant.getParamList().get(2).getName(), is(equalTo("somefile")));
+		assertThat(variant.getParamList().get(2).getValue(), is(equalTo("file.ext")));
+		assertThat(variant.getParamList().get(3).getName(), is(equalTo("somefile")));
+		assertThat(variant.getParamList().get(3).getValue(), is(equalTo("text/plain")));
+	}
+
+	@Test
+	public void shouldInjectParamValueModificationInGeneralParam() {
+		// Given
+		VariantMultipartFormParameters variant = new VariantMultipartFormParameters();
+		HttpMessage message = createMessage();
+		String paramName = "person";
+		String newValue = "injected";
+		String origValue = "anonymous" + CRLF + "anonymous2" + CRLF;
+		variant.setMessage(message);
+		// When
+		variant.setParameter(message,
+				new NameValuePair(NameValuePair.TYPE_POST_DATA, paramName, origValue, 1),
+				paramName, newValue);
+		// Then
+		assertFalse(message.getRequestBody().toString().contains(origValue));
```

Comment:
Better with exact match, that ensures we have the proper contents/format. Same comment for the following tests.